spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/segmentation/MaskUtils.scala

```diff
@@ -61,29 +61,72 @@ object PolyMasks {
  * Also note that we don't use COCO's "compact" RLE string here because this RLE class has better
  * time & space performance.
  *
- * @param counts the RLE counts
  * @param height height of the image
  * @param width width of the image
  */
-class RLEMasks(val counts: Array[Int], val height: Int, val width: Int) extends SegmentationMasks {
-  override def toRLETensor: Tensor[Float] = {
-    Tensor(counts.map(MaskUtils.uint2long(_).toFloat), Array(counts.length))
-  }
+abstract class RLEMasks(val height: Int, val width: Int)
+  extends SegmentationMasks {
+  override def toRLETensor: Tensor[Float]
+
+  // cached bbox value
+  @transient
+  lazy val bbox: (Float, Float, Float, Float) = MaskUtils.rleToOneBbox(this)
+
+  // cached area value
+  @transient
+  lazy val area: Long = MaskUtils.rleArea(this)
 
+  /**
+   * @return The length of counts data
+   */
+  def countsLength: Int
+  def counts: Array[Int]
   /**
    * Get an element in the counts. Process the overflowed int
    *
    * @param idx
    * @return
    */
-  def get(idx: Int): Long = {
-    MaskUtils.uint2long(counts(idx))
+  def get(idx: Int): Long
+}
+
+class RLEMasksInt(private val _counts: Array[Int], height: Int, width: Int)
+  extends RLEMasks(height, width) {
+  override def toRLETensor: Tensor[Float] = {
+    Tensor(_counts.map(MaskUtils.uint2long(_).toFloat), Array(_counts.length))
+  }
+
+  override def countsLength: Int = _counts.length
+
+  override def get(idx: Int): Long = {
+    MaskUtils.uint2long(_counts(idx))
+  }
+
+  override def counts: Array[Int] = _counts
+}
+
+class RLEMasksFloat(private val _counts: Tensor[Float], height: Int, width: Int)
```

Comment:
I don't think we need two types of RLEMask

Reply:
Changed back to old implementation